### PR TITLE
feat(args): parse Sentry web URLs as CLI arguments

### PR DESCRIPTION
**Pull request by BYK**
## Summary

- Users can now paste Sentry web URLs directly as CLI arguments (e.g., `sentry issue view https://sentry.example.com/organizations/my-org/issues/32886/?project=2`) instead of manually extracting org/project/issue IDs
- Supports both SaaS (sentry.io) and self-hosted instances — self-hosted URLs automatically configure `SENTRY_URL` for the session
- Fixes frozen module-level constants in `oauth.ts` and `sentry-client.ts` that prevented dynamic URL detection from working

## URL Patterns Supported

| Pattern | Used By |
|---------|---------|
| `/organizations/{org}/issues/{id}/` | `issue view`, `issue explain`, `issue plan` |
| `/organizations/{org}/issues/{id}/events/{eventId}/` | `event view` |
| `/settings/{org}/projects/{project}/` | `project view`, `project list` |
| `/organizations/{org}/traces/{traceId}/` | `trace view` |
| `/organizations/{org}/` | org-scoped commands |

### New Files
- **`src/lib/sentry-url-parser.ts`** — Pure URL parser (`parseSentryUrl`) + self-hosted env config (`applySentryUrlContext`)
- **`test/lib/sentry-url-parser.test.ts`** — 30 unit tests covering all URL patterns, edge cases, and env var behavior
- **`test/lib/sentry-url-parser.property.test.ts`** — Round-trip property tests verifying URLs built by `sentry-urls.ts` can be parsed back

### Modified Files
- **`src/lib/arg-parsing.ts`** — URL pre-parsing in `parseIssueArg()` and `parseOrgProjectArg()`
- **`src/commands/event/view.ts`** — URL handling in `parsePositionalArgs()` for event URLs
- **`src/lib/oauth.ts`** — Replaced frozen `const SENTRY_URL` with lazy `getSentryUrl()` function
- **`src/lib/sentry-client.ts`** — Removed frozen `const CONTROL_SILO_URL`, made `getControlSiloUrl()` read env var lazily

### Test Updates
- **`test/lib/arg-parsing.test.ts`** — Added URL integration tests for both parsers
- **`test/commands/event/view.test.ts`** — Added event URL handling tests

## Fixes

Resolves CLI-66 (self-hosted users get confusing errors when pasting URLs)

### CODE REVIEW

File: src/lib/sentry-url-parser.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * @param baseUrl - The scheme + host extracted from the URL (e.g., "https://sentry.example.com")
+ */
+export function applySentryUrlContext(baseUrl: string): void {
+  if (isSentrySaasUrl(baseUrl)) {
+    return;
+  }
+  if (process.env.SENTRY_URL) {
+    return;
+  }
```
Comment (BYK) [inline src/lib/sentry-url-parser.ts]: I'd say the parsed URL should take precedence over any previously set env variable.

File: src/lib/sentry-client.ts [hunk 2]
```diff
@@ -12,11 +12,10 @@ import { DEFAULT_SENTRY_URL, getUserAgent } from "./constants.js";
...
 
-/**
- * Control silo URL - handles OAuth, user accounts, and region routing.
- * This is always sentry.io for SaaS, or the base URL for self-hosted.
- */
-const CONTROL_SILO_URL = process.env.SENTRY_URL || DEFAULT_SENTRY_URL;
+// CONTROL_SILO_URL was previously a module-level const that captured
+// process.env.SENTRY_URL at import time. This broke self-hosted URL
+// detection where SENTRY_URL is set after import. Now read lazily
+// via getControlSiloUrl().
```
Comment (BYK) [inline src/lib/sentry-client.ts]:
```suggestion
```
